lib/Driver/ToolChains.h

```diff
@@ -50,6 +50,20 @@ class LLVM_LIBRARY_VISIBILITY GenericUnix : public ToolChain {
   ~GenericUnix() = default;
 };
 
+class LLVM_LIBRARY_VISIBILITY Android : public ToolChain {
```

Comment:
Good call. There's a lot of duplication here! `toolchains::Windows` is also very similar--do you think that should inherit from some more generic ToolChain as well?